File: internal/billing/plans.go
package billing

import "os"

const (
	PlanFree       = "free"
	PlanTeam       = "team"
	PlanPro        = "pro"
	PlanEnterprise = "enterprise"
)

// Entitlements describes what an org is allowed to do under a given plan.
// -1 means unlimited.
type Entitlements struct {
	Plan                string
	MaxMembers          int
	MaxEnvironments     int
	HistoryDays         int
	SlackIntegration    bool
	TeamsIntegration    bool
	KubernetesIngestion bool
	AISummaries         bool
	AIPostmortems       bool
	SSO                 bool
	AuditLogs           bool
}

var planEntitlements = map[string]Entitlements{
	PlanFree: {
		Plan:             PlanFree,
		MaxMembers:       5,
		MaxEnvironments:  2,
		HistoryDays:      30,
		SlackIntegration: true,
		TeamsIntegration: true,
	},
	PlanTeam: {
		Plan:             PlanTeam,
		MaxMembers:       25,
		MaxEnvironments:  -1,
		HistoryDays:      -1,
		SlackIntegration: true,
		TeamsIntegration: true,
		AISummaries:      true,
	},
	PlanPro: {
		Plan:                PlanPro,
		MaxMembers:          -1,
		MaxEnvironments:     -1,
		HistoryDays:         -1,
		SlackIntegration:    true,
		TeamsIntegration:    true,
		KubernetesIngestion: true,
		AISummaries:         true,
		AIPostmortems:       true,
		SSO:                 true,
		AuditLogs:           true,
	},
	PlanEnterprise: {
		Plan:                PlanEnterprise,
		MaxMembers:          -1,
		MaxEnvironments:     -1,
		HistoryDays:         -1,
		SlackIntegration:    true,
		TeamsIntegration:    true,
		KubernetesIngestion: true,
		AISummaries:         true,
		AIPostmortems:       true,
		SSO:                 true,
		AuditLogs:           true,
	},
}

// PricingInfo describes public-facing plan pricing.
type PricingInfo struct {
	ID          string
	Name        string
	PricePerSeat int // USD cents per seat per month; 0 = free; -1 = contact sales
	Description string
	Features    []string
}

var PricingTable = []PricingInfo{
	{
		ID:          PlanFree,
		Name:        "Free",
		Description: "For small teams getting started.",
		Features: []string{
			"Up to 5 members",
			"2 environments",
			"30-day history",
			"Slack & Teams notifications",
			"CLI + webhooks",
		},
	},
	{
		ID:           PlanTeam,
		Name:         "Team",
		PricePerSeat: 1500,
		Description:  "For growing engineering teams.",
		Features: []string{
			"Up to 25 members",
			"Unlimited environments",
			"Unlimited history",
			"AI event summaries",
			"All Free features",
		},
	},
	{
		ID:           PlanPro,
		Name:         "Pro",
		PricePerSeat: 3500,
		Description:  "For teams running production at scale.",
		Features: []string{
			"Unlimited members",
			"Kubernetes auto-ingestion",
			"AI postmortems",
			"SSO",
			"Audit logs",
			"All Team features",
		},
	},
	{
		ID:           PlanEnterprise,
		Name:         "Enterprise",
		PricePerSeat: -1,
		Description:  "Custom contracts, SLAs, and dedicated support.",
		Features: []string{
			"SAML / SCIM",
			"Custom data retention",
			"Data residency",
			"Dedicated support",
			"All Pro features",
		},
	},
}

// EntitlementsFor returns the entitlements for a given plan.
// When SELF_HOSTED=true all features are unlocked regardless of plan.
func EntitlementsFor(plan string) Entitlements {
	if os.Getenv("SELF_HOSTED") == "true" {
		return planEntitlements[PlanEnterprise]
	}
	if e, ok := planEntitlements[plan]; ok {
		return e
	}
	return planEntitlements[PlanFree]
}

File: ui/app/pricing/plans.ts
/**
 * Plan data for the pricing page.
 *
 * Extracted from page.tsx so the route's structured data (layout.tsx) is built
 * from the same source the table renders from — schema prices that drift from
 * displayed prices are a rich-result violation.
 */

export type Plan = {
  id: string
  name: string
  price: string
  per: string
  description: string
  cta: string
  ctaVariant: "default" | "outline"
  highlight: boolean
  badge?: string
  features: string[]
}

export const PLANS: Plan[] = [
  {
    id: "free",
    name: "Free",
    price: "$0",
    per: "forever",
    description: "For small teams getting started.",
    cta: "Get started",
    ctaVariant: "outline",
    highlight: false,
    features: [
      "Up to 5 members",
      "2 environments",
      "30-day event history",
      "Webhook ingestion",
      "CLI tool",
      "Email notifications",
    ],
  },
  {
    id: "team",
    name: "Team",
    price: "$12",
    per: "seat/mo",
    description: "Growing teams that need more room.",
    cta: "Start free trial",
    ctaVariant: "outline",
    highlight: false,
    features: [
      "Up to 25 members",
      "Unlimited environments",
      "Unlimited history",
      "AI event summaries",
      "AI timeline assistant",
      "Slack & Teams notifications",
      "Email support",
    ],
  },
  {
    id: "pro",
    name: "Pro",
    price: "$24",
    per: "seat/mo",
    description: "Engineering orgs that run on Kubernetes.",
    cta: "Start free trial",
    ctaVariant: "default",
    highlight: true,
    badge: "Most popular",
    features: [
      "Unlimited members",
      "Unlimited environments",
      "Unlimited history",
      "AI summaries + postmortems",
      "AI timeline assistant",
      "Kubernetes watcher ingestion",
      "SSO (SAML / OIDC)",
      "Audit logs",
      "Priority support",
    ],
  },
  {
    id: "enterprise",
    name: "Enterprise",
    price: "Custom",
    per: "per org",
    description: "Dedicated support and custom contracts.",
    cta: "Contact us",
    ctaVariant: "outline",
    highlight: false,
    features: [
      "Everything in Pro",
      "Volume seat pricing",
      "Dedicated Slack channel",
      "SLA guarantees",
      "Custom data retention",
      "On-prem deployment option",
    ],
  },
]
